docs/lsm/allocation/allocation_v3.rst

```diff
@@ -0,0 +1,175 @@
+**************
+Allocation V3
+**************
+
+
+Allocation V3 is a new framework that changes significantly compared to Allocation V2. It doesn't have
+its own lifecycle stage anymore, but instead allocation happens during validation compile. The purpose is
```

Comment:
> It doesn't have its own lifecycle stage anymore, but instead allocation happens during validation compile.

That is not true, nothing changed lifecycle-wise.  For any type of allocation, there needs to be a validating state, in which a validation compile is run.